plugins/duo_admin/komand_duo_admin/tasks/monitor_logs/task.py

```diff
@@ -28,96 +31,155 @@ def __init__(self):
-                last_24_hours = now - timedelta(hours=24)
-                mintime_in_milliseconds = self.convert_to_milliseconds(last_24_hours)
-                maxtime_in_milliseconds = self.convert_to_milliseconds(last_minute)
-                mintime_in_seconds = self.convert_to_seconds(last_24_hours)
-                maxtime_in_seconds = self.convert_to_seconds(last_minute)
+            if last_collection_timestamp:
+                # Previously only one timestamp was held (the end of the collection window)
+                # This has been superceded by a latest timestamp per log type
+                self.logger.info("Backwards compatibility - update all timestamps to the last known timestamp")
+                trust_monitor_last_log_timestamp = (
+                    auth_logs_last_log_timestamp
+                ) = admin_logs_last_log_timestamp = last_collection_timestamp
+                # Update the old last collection timestamp to None so it is not considered in future runs
+                state[self.LAST_COLLECTION_TIMESTAMP] = None
             else:
-                if trust_monitor_next_page_params or auth_logs_next_page_params or admin_logs_next_page_params:
-                    self.logger.info("Getting the next page of results...")
-                    get_next_page = True
-                else:
-                    self.logger.info("Subsequent run")
-                mintime_in_milliseconds = last_collection_timestamp
-                maxtime_in_milliseconds = self.convert_to_milliseconds(last_minute)
-                mintime_in_seconds = int(mintime_in_milliseconds / 1000)
-                maxtime_in_seconds = int(maxtime_in_milliseconds / 1000)
+                trust_monitor_last_log_timestamp = state.get(self.TRUST_MONITOR_LAST_LOG_TIMESTAMP)
+                auth_logs_last_log_timestamp = state.get(self.AUTH_LOGS_LAST_LOG_TIMESTAMP)
+                admin_logs_last_log_timestamp = state.get(self.ADMIN_LOGS_LAST_LOG_TIMESTAMP)
+                self.logger.info(
+                    f"Previous timestamps retrieved. "
+                    f"Auth {auth_logs_last_log_timestamp}. "
+                    f"Admin: {admin_logs_last_log_timestamp}. "
+                    f"Trust monitor {trust_monitor_last_log_timestamp}."
+                )
             try:
                 new_logs = []
-                if not get_next_page:
-                    state[self.LAST_COLLECTION_TIMESTAMP] = maxtime_in_milliseconds
 
                 previous_trust_monitor_event_hashes = state.get(self.PREVIOUS_TRUST_MONITOR_EVENT_HASHES, [])
                 previous_admin_log_hashes = state.get(self.PREVIOUS_ADMIN_LOG_HASHES, [])
                 previous_auth_log_hashes = state.get(self.PREVIOUS_AUTH_LOG_HASHES, [])
-                new_trust_monitor_event_hashes = []
-                new_admin_log_hashes = []
-                new_auth_log_hashes = []
+
+                new_trust_monitor_event_hashes = new_admin_log_hashes = new_auth_log_hashes = []
```

Comment:
I think this creates a reference between each variable, rather than setting each value to `[]`? I think the state hashes at the end will all equal the same value? should it not be:

```
new_trust_monitor_event_hashes = new_admin_log_hashes = new_auth_log_hashes = [], [], []
```

Reply:
but change it for a,b,c = [], [], []

Reply:
As above 😄 

```
new_trust_monitor_event_hashes, new_admin_log_hashes, new_auth_log_hashes = [], [], []
```

Reply:
My bad copying and pasting and then updating 🤦